pkgs/development/compilers/solc/default.nix

```diff
@@ -1,28 +1,30 @@
 { stdenv, fetchzip, fetchgit, boost, cmake }:
 
+let rev = "194ff033ae44944ac59aa7bd3da89ba94ec5893c"; in
+let sha256 = "0gkg3nay0625qmhxxxax1d1c4dl554ri3pkwd12qfg6g1w6j04w7"; in
+let version = "0.4.12"; in
+
+let jsoncppURL = https://github.com/open-source-parsers/jsoncpp/archive/1.7.7.tar.gz; in
 let jsoncpp = fetchzip {
-  url = https://github.com/open-source-parsers/jsoncpp/archive/1.7.7.tar.gz;
+  url = jsoncppURL;
   sha256 = "0jz93zv17ir7lbxb3dv8ph2n916rajs8i96immwx9vb45pqid3n0";
 }; in
 
-let commit = "68ef5810593e7c8092ed41d5f474dd43141624eb"; in
-
-stdenv.mkDerivation rec {
-  version = "0.4.11";
+stdenv.mkDerivation {
+  inherit version;
   name = "solc-${version}";
 
   # Cannot use `fetchFromGitHub' because of submodules
   src = fetchgit {
     url = "https://github.com/ethereum/solidity";
-    rev = commit;
-    sha256 = "13zycybf23yvf3hkf9zgw9gbc1y4ifzxaf7sll69bsn24fcyq961";
+    inherit rev sha256;
   };
 
   patchPhase = ''
-    echo >commit_hash.txt ${commit}
+    echo >commit_hash.txt ${rev}
     echo >prerelease.txt
     substituteInPlace deps/jsoncpp.cmake \
-      --replace https://github.com/open-source-parsers/jsoncpp/archive/1.7.7.tar.gz ${jsoncpp}
+      --replace ${jsoncppURL} ${jsoncpp}
```

Comment:
@dbrock There are two levels of code injection possible. One is in the builder (it should be safe to modify the URL to any value) and the other is in CMake. 

I am not sure whether this level of correctness is actually achieved by the rest of NixPkgs and assuming you are just interested in getting this into NixPkgs, I am not the one holding back this PR, in case you were thinking that.